core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/DiagnosticLog.java

```diff
@@ -172,62 +184,14 @@ public static class DiagnosticLogBuilder {
         private final String componentId;
         private Map<String, Object> input;
         private Map<String, Object> configurations;
-        private LogLevel logLevel;
+        private LogDetailLevel logDetailLevel;
 
         public DiagnosticLogBuilder(String componentId, String actionId) {
 
             this.componentId = componentId;
             this.actionId = actionId;
         }
 
-        /**
-         * Sets the log ID of the DiagnosticLog.
-         *
-         * @param logId the log ID to be set.
-         * @return the current DiagnosticLogBuilder instance.
-         */
-        public DiagnosticLogBuilder logId(String logId) {
-
-            this.logId = logId;
-            return this;
-        }
-
-        /**
-         * Sets the recordedAt timestamp of the DiagnosticLog.
-         *
-         * @param recordedAt the timestamp to be set.
-         * @return the current DiagnosticLogBuilder instance.
-         */
-        public DiagnosticLogBuilder recordedAt(Instant recordedAt) {
-
-            this.recordedAt = recordedAt;
-            return this;
-        }
-
-        /**
-         * Sets the request ID of the DiagnosticLog.
-         *
-         * @param requestId the request ID to be set.
-         * @return the current DiagnosticLogBuilder instance.
-         */
-        public DiagnosticLogBuilder requestId(String requestId) {
-
-            this.requestId = requestId;
-            return this;
-        }
-
-        /**
-         * Sets the flow ID of the DiagnosticLog.
-         *
-         * @param flowId the flow ID to be set.
-         * @return the current DiagnosticLogBuilder instance.
-         */
-        public DiagnosticLogBuilder flowId(String flowId) {
```

Comment:
why were these removed?

Reply:
In the current implementation, we are getting the values for this using the MDC. Currently, the build() method by default set these attributes. so there's no point of introducing these setters

Reply:
ok, then we should get rid of the if checks at the build, right? they will always be null/empty now?

Reply:
Fixed with b5a4314b5e95e7a7e26ac262ffd81f8f79a3d408